indra/llcommon/llcoros.cpp

```diff
@@ -297,7 +297,8 @@ std::string LLCoros::launch(const std::string& prefix, const callable_t& callabl
         // Out of memory on stack allocation?
         LLError::LLUserWarningMsg::showOutOfMemory();
         printActiveCoroutines();
-        LL_ERRS("LLCoros") << "Bad memory allocation in LLCoros::launch(" << prefix << ")!" << LL_ENDL;
+        LL_WARNS("LLCoros") << "Bad memory allocation in LLCoros::launch(" << prefix << ")!" << LL_ENDL;
+        LLApp::quit();
```

Comment:
I stipulate that in this situation, the viewer has always before just terminated without explanation. But if we're trying to improve that situation, maybe put up a notification? It seems worth a try: inability to allocate a contiguous block of `mStackSize` bytes doesn't mean it would be impossible to allocate something smaller.

As coded, this new `quit()` call will call `LLApp::requestQuit()`. Unless the user has previously clicked "I never want to see this confirmation prompt again," `requestQuit()` will pop up a message asking the user if they're sure they want to quit. As a user, I would be confused by that prompt, and would click "No, of course I didn't ask to quit!" That would probably return from `quit()` (I haven't tested this scenario), leading to internal confusion and a likely crash.

I would rather see a modal popup saying something about the out-of-memory condition and apologetically informing the user that the viewer must shut down -- much like the `LL_ERRS()` popup in a developer build. (Maybe reuse that notification?) On return, we could unconditionally call `forceQuit()`.

Reply:
Viewer currently calls showOutOfMemory(); which informs user of what's going on.

Reply:
Thank you, I did overlook that.

I still think it would be odd to then put up a second prompt asking the user if they really want to quit, and returning normally if they click No.

Reply:
> Viewer currently calls showOutOfMemory() at string 298 which informs user of what's going on.
Sure, this looks as follows:
![image](https://github.com/user-attachments/assets/c9b984fc-4c5a-4d64-a909-1f2b8596e6e2)

Reply:
> Unless the user has previously clicked "I never want to see this confirmation prompt again," `requestQuit()` will pop up a message asking the user if they're sure they want to quit

I simulate this error, and I observe only one message box (shown above).
After clicking "OK" button I see the viewer is terminated silently in a few seconds.

Reply:
requestQuit doesn't show any notifications, it just does a sendLogoutRequest then forceQuit